src/cosl/coordinated_workers/nginx.py

```diff
@@ -29,6 +31,319 @@
     "nginx_port": 8080,
     "nginx_exporter_port": 9113,
 }
+RESOLV_CONF_PATH = "/etc/resolv.conf"
+
+
+class NginxLocationModifier(Enum):
+    """Enum representing valid Nginx `location` block modifiers."""
+
+    DEFAULT = ""  # prefix match
+    EXACT = "="  # exact match
+    REGEX = "~"  # case-sensitive regex
+
+
+@dataclass
+class NginxLocationConfig:
+    """Represents a `location` block in an Nginx configuration."""
+
+    upstream: str
+    path: str = "/"
+    modifier: NginxLocationModifier = NginxLocationModifier.DEFAULT
+    is_grpc: bool = False
+
+
+class NginxConfig:
+    """Builds the Nginx configuration."""
+
+    def __init__(
+        self,
+        server_name: str,
+        roles_to_upstreams: Dict[str, Dict[str, int]],
```

Comment:
I'd add a docstring with some clarifications about the content of these vars and an example. These data structures are quite packed with info.